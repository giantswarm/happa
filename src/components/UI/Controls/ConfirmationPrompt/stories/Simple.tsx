import { StoryFn } from '@storybook/react';
import { Box, Text } from 'grommet';
import React, { ComponentPropsWithoutRef, useState } from 'react';
import Button from 'UI/Controls/Button';

import ConfirmationPrompt from '..';

export const Simple: StoryFn<
  ComponentPropsWithoutRef<typeof ConfirmationPrompt>
> = (args) => {
  const [open, setIsOpen] = useState(false);

  return (
    <Box>
      <Box margin={{ bottom: 'small' }}>
        <Button onClick={() => setIsOpen(!open)}>Open prompt</Button>
      </Box>
      <ConfirmationPrompt {...args} open={open}>
        <Text>Check out this awesome component!</Text>
      </ConfirmationPrompt>
    </Box>
  );
};
