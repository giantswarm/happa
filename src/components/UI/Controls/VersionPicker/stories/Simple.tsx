import { action } from '@storybook/addon-actions';
import { Story } from '@storybook/react/types-6-0';
import React, { ComponentPropsWithoutRef } from 'react';

import VersionPicker from '../VersionPicker';

export const Simple: Story<ComponentPropsWithoutRef<typeof VersionPicker>> = (
  args
) => {
  return (
    <VersionPicker
      onChange={action('changed')}
      selectedVersion={args.selectedVersion}
      versions={[
        {
          created: '2020-12-15T16:39:27Z',
          includesVersion: '2.0.0',
          chartVersion: '1.0.5',
          test: false,
        },
        {
          created: '2020-12-14T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '1.0.3',
          test: false,
        },
        {
          created: '2020-12-13T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '1.0.2',
          test: false,
        },
        {
          created: '2020-12-12T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.9',
          test: false,
        },
        {
          created: '2020-12-11T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.8',
          test: false,
        },
        {
          created: '2020-12-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.7',
          test: false,
        },
        {
          created: '2020-11-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.6',
          test: false,
        },
        {
          created: '2020-10-10T16:39:27Z',
          includesVersion: '1.0.0',
          chartVersion: '0.8.5',
          test: false,
        },
        {
          created: '2020-09-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.4',
          test: false,
        },
        {
          created: '2020-08-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.3',
          test: false,
        },
        {
          created: '2020-07-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.2',
          test: false,
        },
        {
          created: '2020-06-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.1',
          test: false,
        },
        {
          created: '2020-05-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.8.0',
          test: false,
        },
        {
          created: '2020-04-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.10',
          test: false,
        },
        {
          created: '2020-03-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.8',
          test: false,
        },
        {
          created: '2020-02-10T16:39:27Z',
          includesVersion: '0.0.0',
          chartVersion: '0.7.3',
          test: false,
        },
      ]}
    />
  );
};

Simple.args = {
  selectedVersion: '1.0.2',
};

Simple.argTypes = {
  selectedVersion: {
    description: 'The version selected when rendering the component',
    control: { type: 'text' },
  },
};
