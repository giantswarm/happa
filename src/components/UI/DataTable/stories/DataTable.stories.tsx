import { Meta } from '@storybook/react';

import DataTable from '..';

export { Simple } from './Simple';

export default {
  title: 'Display/DataTable',
  component: DataTable,
} as Meta;
