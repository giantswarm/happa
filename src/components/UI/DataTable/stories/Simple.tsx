import { Story } from '@storybook/react';
import { DataTableProps, Text } from 'grommet';
import React from 'react';

import DataTable from '../';

interface IData {
  name: string;
  location: string;
  amount: number;
}

export const Simple: Story<DataTableProps> = (args) => {
  return <DataTable<IData> {...args} />;
};

Simple.args = {
  columns: [
    {
      property: 'name',
      header: 'Name',
      primary: true,
    },
    {
      property: 'location',
      header: 'Location',
      sortable: false,
    },
    {
      property: 'amount',
      header: 'Amount',
      render: (data) => <Text>${data.amount}</Text>,
    },
  ],
  data: [
    {
      name: 'Food',
      location: 'Amazon',
      amount: 25.36,
    },
    {
      name: 'Laptop',
      location: 'Apple',
      amount: 1683.0,
    },
    {
      name: 'Phone',
      location: 'Apple',
      amount: 1037.73,
    },
  ],
  sortable: true,
};

Simple.argTypes = {};
