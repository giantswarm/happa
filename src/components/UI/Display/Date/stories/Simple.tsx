import { Story } from '@storybook/react';
import React from 'react';

import Date from '..';

export const Simple: Story<React.ComponentPropsWithoutRef<typeof Date>> = (
  args
) => {
  return (
    <div>
      <Date {...args} />
    </div>
  );
};

Simple.args = {};

Simple.argTypes = {
  value: {
    control: {
      type: 'date',
    },
    defaultValue: '2021-01-10',
    name: 'date (local time zone)',
  },
};
