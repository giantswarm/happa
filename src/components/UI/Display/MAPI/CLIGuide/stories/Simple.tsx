import { Story } from '@storybook/react';
import { Box, Text } from 'grommet';
import React from 'react';
import { CodeBlock, Prompt } from 'UI/Display/Documentation/CodeBlock';

import CLIGuide from '..';

export const Simple: Story<React.ComponentPropsWithoutRef<typeof CLIGuide>> = (
  args
) => {
  return <CLIGuide {...args} />;
};

Simple.args = {
  title: 'Get some data from the somewhere',
  children: (
    <Box direction='column' gap='small'>
      <Box direction='column' gap='small'>
        <Text>1. Make sure you are a boss</Text>
        <CodeBlock>
          <Prompt>some command --some-flag</Prompt>
        </CodeBlock>
      </Box>
      <Box direction='column' gap='small'>
        <Text>2. Maybe check again if you are still a boss</Text>
        <CodeBlock>
          <Prompt>some-other-cli checkboss</Prompt>
        </CodeBlock>
      </Box>
    </Box>
  ),
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
