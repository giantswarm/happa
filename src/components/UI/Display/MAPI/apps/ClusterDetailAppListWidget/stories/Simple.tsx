import { Story } from '@storybook/react';
import { Box, Heading, Text } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailAppListWidget from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailAppListWidget>
> = (args) => {
  return (
    <ClusterDetailAppListWidget {...args}>
      <Text>Hi everyone. This is a widget that you can customize</Text>
      <Box direction='row' pad='small' gap='small' margin={{ top: 'medium' }}>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
        <Box align='center'>
          <Heading level={3} margin='none'>
            50
          </Heading>
          <Text>shades of happa</Text>
        </Box>
      </Box>
    </ClusterDetailAppListWidget>
  );
};

Simple.args = {
  title: 'Some widget',
};

Simple.argTypes = {
  title: { control: { type: 'text' } },
};
