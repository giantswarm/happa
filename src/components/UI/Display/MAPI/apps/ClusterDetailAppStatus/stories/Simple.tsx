import { Story } from '@storybook/react';
import { Box } from 'grommet';
import React, { ComponentPropsWithoutRef } from 'react';

import ClusterDetailAppStatus from '..';

export const Simple: Story<
  ComponentPropsWithoutRef<typeof ClusterDetailAppStatus>
> = (args) => {
  return (
    <Box gap='medium' align='flex-start'>
      <ClusterDetailAppStatus {...args} />
    </Box>
  );
};

Simple.args = {
  status: 'some-status',
};

Simple.argTypes = {
  status: { control: { type: 'text' } },
};
