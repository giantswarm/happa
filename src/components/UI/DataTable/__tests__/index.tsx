import { renderWithTheme } from 'testUtils/renderUtils';

import { Simple } from '../stories/Simple';

describe('DataTable', () => {
  it('renders a simple table', () => {
    const { container } = renderWithTheme(Simple, {});
    expect(container.firstChild).toMatchSnapshot();
  });
});
