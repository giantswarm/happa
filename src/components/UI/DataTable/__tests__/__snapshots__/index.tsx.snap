// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable renders a simple table 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 iUQqn"
>
  <div
    class="sc-gKAaRy ezfMqp"
  >
    <table
      class="StyledTable-sc-1m3u5g-6 gnELFm StyledDataTable-xrlyjm-0 efGflZ"
    >
      <thead
        class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-xrlyjm-4"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-xrlyjm-2 jcwRQa"
        />
      </thead>
      <tbody
        class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-xrlyjm-3 hIVAIv"
      />
    </table>
  </div>
</div>
`;
